File: Code Java/Java_Duc/src/Phanso/phanso.java

package Phanso;

public class phanso {
private
        int tuso, mauso;
public phanso(){}
public phanso(int tuso, int mauso){
        this.tuso = tuso;
        this.mauso = mauso;
    }
public int gcd(int tuso, int mauso){
    int temp = 0;
    while(mauso != 0){
        temp = tuso % mauso;
        tuso = mauso;
        mauso = temp;
    }
    return tuso;
}
public void rutgon() {
    int gcd = gcd(tuso, mauso);
    tuso = tuso / gcd;
    mauso = mauso / gcd;
    }
public phanso cong(phanso a, phanso b){
    phanso kq = new phanso();
    kq.tuso = a.tuso * b.mauso + b.tuso * a.mauso;
    kq.mauso  = a.mauso * b.mauso;
    kq.rutgon();
    return kq;
}
public phanso hieu(phanso a, phanso b){
    phanso kqhieu = new phanso();
    kqhieu.tuso = a.tuso * b.mauso - b.tuso * a.mauso;
    kqhieu.mauso  = a.mauso * b.mauso;
    kqhieu.rutgon();
    return kqhieu;
}
public phanso tich(phanso a, phanso b){
    phanso kqtich = new phanso();
    kqtich.tuso = a.tuso * b.tuso;
    kqtich.mauso = a.mauso * b.mauso;
    kqtich.rutgon();
    return kqtich;
}
public phanso thuong(phanso a, phanso b){
    phanso kqthuong = new phanso();
    kqthuong.tuso = a.tuso * b.mauso;
    kqthuong.mauso = a.mauso * b.tuso;
    kqthuong.rutgon();
    return kqthuong;
}

public void xuat(phanso a){
    System.out.println(""+ a.tuso + "/" + a.mauso);
}
}

File: Code C++/Trenlop/Phanso.cpp
#include <iostream>
using namespace std;
class PhanSo{
    protected:
        int tuso, mauso;
    public:
        PhanSo(){
            tuso = 0;
            mauso = 1;
        }
        PhanSo(int tuso, int mauso){
            this->mauso = mauso;
            this->tuso = tuso;
        }
        
        friend istream &operator >> (istream &is, PhanSo &sp1){
            is >> sp1.tuso >> sp1.mauso;
            return is;
        }
        friend ostream &operator << (ostream &os,PhanSo sp1){
            os << sp1.tuso << "/" << sp1.mauso;
            return os;
        }
        int gcd(int a, int b) {
            while (b != 0) {
                int temp = a % b;
                a = b;
                 b = temp;
            }
            return a;
        }
        PhanSo rutgon() {
            int ucln = gcd(tuso, mauso);
            return PhanSo(tuso / ucln, mauso / ucln);
        }
        PhanSo operator + (PhanSo sp1){
            PhanSo s;
            s.tuso = tuso * sp1.mauso + mauso * sp1.tuso;
            s.mauso = mauso * sp1.mauso;
            return s;
        }

        PhanSo operator - (PhanSo sp1){
            PhanSo m;
            m.tuso = tuso * sp1.mauso - mauso * sp1.tuso;
            m.mauso = mauso * sp1.mauso;
            return m;
        }
        PhanSo operator * (PhanSo sp1){
            PhanSo n;
            n.tuso = tuso * sp1.tuso;
            n.mauso = mauso * sp1.mauso;
            return n;
        }
        PhanSo operator / (PhanSo sp1){
            PhanSo s;
            s.tuso = tuso * sp1.mauso;
            s.mauso = mauso * sp1.tuso;
            return s;
        }
        
};
class PhanSo2 : public PhanSo{
    public:
        PhanSo2() : PhanSo() {}
        PhanSo2(int tuso, int mauso) : PhanSo(tuso, mauso) {}

        PhanSo2 operator = (PhanSo2 sp1){
            PhanSo2 sp2;
           sp2.mauso = sp1.mauso;
           sp2.tuso = sp1.tuso;
           return sp2;
        }
        
};
int main(){
    int n;
    cout << "Nhap vao so luong phan so: ";
    cin >> n;
    PhanSo2 sp2[10];
    for(int i = 0; i< n; i++){
        cin >> sp2[i];
    }
    for(int i = 0; i< n; i++){
        cout <<sp2[i];
    }
    PhanSo sum;   
    for(int i = 0; i < n;i++){
     sum = sum + sp2[i];
    }
    cout << "Tong cac phan so: " << sum.rutgon() << endl;
}